operations_on_datastructures/trie_multiple_search.cpp

```diff
@@ -0,0 +1,458 @@
+/**
+ * @file
+ * @brief Trie datastructure with search variants

```

Comment:
Provide a Wikipedia link in Markdown format (if possible/available).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂